firebase-app-distribution/src/main/java/com/google/firebase/app/distribution/NewReleaseFetcher.java

```diff
@@ -160,7 +160,15 @@ private boolean canInstall(AppDistributionReleaseInternal newRelease)
   private boolean isNewerBuildVersion(AppDistributionReleaseInternal newRelease)
       throws FirebaseAppDistributionException {
     return Long.parseLong(newRelease.getBuildVersion())
-        > getInstalledAppVersionCode(firebaseApp.getApplicationContext());
+            > getInstalledAppVersionCode(firebaseApp.getApplicationContext())
+        || !doesAppVersionNameMatch(newRelease);
+  }
+
+  private boolean doesAppVersionNameMatch(AppDistributionReleaseInternal newRelease)
```

Comment:
To avoid having to negate the return value where this is used, and to improve readability, I'd change this to be `hasDifferentAppVersion` and return the opposite boolean value.

Reply:
ok